torch_xla/experimental/eager.py

```diff
@@ -1,9 +1,48 @@
+import functools
+
 import torch_xla
 
 
 def eager_mode(enable: bool):
   """Configure torch_xla's default executation mode.
+
   Under eager mode only functions that was `torch_xla.compile`d will be
   traced and compiled. Other torch ops will be executed eagerly.
   """
   torch_xla._XLAC._set_use_eager_mode(enable)
+
+
+def is_eager_mode() -> bool:
+  """Return True if torch_xla is currently under eager mode
+  """
+  return torch_xla._XLAC._get_use_eager_mode()
+
+
+def compile(func):
+  """Compile the func with Lazy Tensor.
+
+  Return the optimized function that takes exact same input. Compile will
+  run the target func under the tracing mode using Lazy tensor.
+  """
+
+  @functools.wraps(func)  # Keep function's name, docstring, etc.
+  def wrapper(*args, **kwargs):
+    # compile should only be called with
+    assert torch_xla._XLAC._get_use_eager_mode() == True
+    torch_xla._XLAC._set_use_eager_mode(False)
+    # clear the pending graph if any
+    torch_xla.sync()
+    try:
+      # Target Function Execution
+      result = func(*args, **kwargs)
+    except Exception as e:
+      # Handle exceptions (if needed)
+      print(f"Error in target function: {e}")
+      raise  # Re-raise the exception
+    # Sync the graph generated by the target function.
+    torch_xla.sync()
```

Comment:
there actaully runs the graph and you might get exceptions here too.

Reply:
right, the way that LTC works is that async execution happens in a separate thread and the runtime error will be set in the unlocker. Next time when we try to get the device lock it will find that exception and throw
https://github.com/pytorch/xla/blob/master/torch_xla/csrc/xla_graph_executor.cpp#L861-L864

That being said I agree with you that there is no harm to put `sync` in the `try` region. Let me update that in the following pr.